### ISSUE TITLE
Integration tests failing in master branch

### ISSUE THREAD

**Issue by jad-hamza** [epfl-lara/stainless#113]
I'm getting some unexpected outputs and some failures when running `sbt test it:test`.

> exception occurred while compiling /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/util/Random.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/annotation/isabelle.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/annotation/package.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/Map.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/Option.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/Either.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/Bag.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/Rational.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/StrOps.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/Set.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/Real.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/StaticChecks.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/package.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/synthesis/Oracle.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/lang/synthesis/package.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/io/StdIn.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/io/FileInputStream.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/io/StdOut.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/io/FileOutputStream.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/io/package.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/math/package.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/collection/List.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/proof/Internal.scala, /home/user/stainless/frontends/stainless-dotty/target/scala-2.11/classes/stainless/proof/package.scala, /tmp/stainless6116734696740130813.scala   

This file contains:

```scala
object Shift1 {
 def foo(i: Byte, j: Long) = i << j
}
```

When running with stainless I get this error.

> [ Error  ] /tmp/stainless6116734696740130813.scala:2:35: Unsupported shift

It looks like this error is normal because the operation is not supported, but the exception is perhaps not caught properly during the tests? There are other exceptions like that in the tests.

I also have failures, such as:
> [info] -  85: verification/valid/Heaps solvr=smt-z3 lucky=false check=true *** FAILED *** (1 second, 124 milliseconds)
> [info]   inox.package$FatalError: Unexpected error from z3 solver: Solver encountered exception: smtlib.parser.Parser$UnexpectedEOFException: Unexpected end of file. Expected: []

This is maybe related to https://github.com/epfl-lara/inox/issues/32

**Comment by jad-hamza** [comment]
I can't reproduce this last issue with Heaps.scala, it only appears from time to time apparently.

**Comment by jad-hamza** [comment]
I also just got a NullPointerException in the ImperativeSuite for Array7.scala: https://gist.github.com/jad-hamza/9a818c035ad3f20d14ba0e78a93d74d8
But I wasn't able to reproduce on a second run

**Comment by mantognini** [comment]
I think the "exception occurred while compiling" error are emitted by dotty itself.

**Comment by samarion** [comment]
As @mantognini said, the messages are output by dotty itself and avoiding the output would require taking a closer look at dotty reporting. I expect these exceptions to transform into nicer reports at some point thanks to ongoing dotty development so I don't think there's much point in us looking into this.

I have never seen your smt-z3 error for Heaps.scala. However, z3 was non-deterministically but fairly consistently returning invalid outputs for MergeSort2.scala (and thus crashing Stainless) at some point so I ignored it in the tests. If the Heaps.scala issue becomes recurrent, we can ignore it as well.

About the NullPointerException, this may be due to parallelism. I've made many changes in Inox (pushed) and Stainless (not yet pushed) to address various issues we had due to now invalid single-thread assumptions. I believe one of these issues could result in a NullPointerExceptions in computeType, so it may now be fixed (in Inox, not yet in Stainless).
